app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchTEIViewModel.kt

```diff
@@ -356,8 +425,17 @@ class SearchTEIViewModel(
         } ?: false
     }
 
+    fun mapDataFetched() {
+        SearchIdlingResourceSingleton.decrement()
+    }
+
     @VisibleForTesting(otherwise = VisibleForTesting.NONE)
     fun setCurrentProgram(program: Program) {
```

Comment:
Why do you need this function for testing?, couldn't you mock `whenever(searchRepository.getProgram(initialProgramUid)) doReturn ...` ?